src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -755,7 +756,7 @@ private void AddAllocatedEndpointInfo(IEnumerable<AppResource> resources)
 
                 sp.EndpointAnnotation.AllocatedEndpoint = new AllocatedEndpoint(
                     sp.EndpointAnnotation,
-                    "localhost",
+                    sp.EndpointAnnotation.TargetHost.Replace("0.0.0.0", Dns.GetHostName()).Replace("[::]", Dns.GetHostName()).Replace("+", Dns.GetHostName()).Replace("*", Dns.GetHostName()),
```

Comment:
Dns.GetHostName() is a synchronous call that can block. Feels like we should move this elsewhere and make it resilient to timeouts and fallback to localhost if it fails.